### DISCUSSION TITLE
Does using `uv_async_send` to two different `uv_async_t`s have guaranteed order?

### DISCUSSION THREAD

**Discussion by KevinEady** [libuv/libuv#3915] [Q&A]
Hi all,

I have two different `uv_async_t`s:

```cpp
uv_async_t first;
uv_async_t second;
```

If I do a  `uv_async_send(&first)` and then `uv_async_send(&second)`, is it guaranteed that `first` callback will be invoked before `second`? Or is ordering not guaranteed?

Thanks, Kevin

**Comment by vtjnash** [comment]
There is no ordering guarantee
